sdks/python/apache_beam/pvalue.py

```diff
@@ -509,6 +510,33 @@ def _side_input_data(self):
         self._input_element_coder())
 
 
+class AsMultiMap(AsSideInput):
+  """Marker specifying a PCollection to be used as an indexable side input.
+
+  Similar to AsDict, but multiple values may be associated per key, and
+  the keys are fetched lazily rather than all having to fit in memory.
+
+  Intended for use in side-argument specification---the same places where
+  AsSingleton and AsIter are used, but returns an interface that allows
+  key lookup.
+  """
+
+  @staticmethod
+  def _from_runtime_iterable(it, options):
+    # Legacy implementation.
+    result = collections.defaultdict(list)
+    for k, v in it:
+      result[k].append(v)
+    return result
```

Comment:
Do you think a defaultdict will cause confusions by returning an empty list for non-existent keys?

Reply:
Its a pain to keep track of present and empty vs non-existent so I would highly suggest to go with the default always being the empty list whether the key exists or not since you'll have to write a bunch of additional logic since the Fn API multimap view currently treats all non-existent keys as empty lists.